### PR TITLE
Cap normalization at 10X upper bound, and 1X lower bound

### LINKED ISSUE #324: Add min/max values for normalization
**Is your feature request related to a problem? Please describe.**
The current normalization notebook provides a warning if an extreme normalization value is present, but still applies this value. This can result in non-sensical normalization

**Describe the solution you'd like**
We should have a cap for what level of normalization can be applied. Specifically, any normalization value that will result in more than 10x increase in signal should be capped at a 10x increase. Similarly, any normalization value that will result in a decrease in signal should be capped at no change in overall intensity.

### PR DESCRIPTION
**Pull request by alex-l-kong**
**What is the purpose of this PR?**

Closes #324. Prevents "ultra-extreme" normalization that increases signal by 10X or decreases it.

**How did you implement your changes**

In `normalize_fov`, add two new masks:

- `abnormal_increase_mask`: checks if any `norm_vals < 0.1`, if so then cap those at `0.1`
- `decrease_mask`: checks if any `norm_vals > 1`, if so then cap at 1

`extreme_mask` should now only consider channels not already fixed by `abnormal_increase_mask` and `decrease_mask`

### CODE REVIEW

**Review by srivarra — APPROVED** [review]
sicko mode

File: src/toffy/normalize.py [hunk 1]
```diff
@@ -718,12 +718,35 @@ def normalize_fov(img_data, norm_vals, norm_dir, fov, channels, extreme_vals):
...
+    if len(decrease_mask) > 0:
+        decrease_chans = np.array(channels)[decrease_mask]
+        warnings.warn(
+            "The following channel(s) will suffer a decrease after normalization "
+            "for fov {}. Skipping normalization for: {}.".format(fov, decrease_chans)
+        )
+        norm_vals[decrease_mask] = 1.0
+
+    # check if any additional values are outside expected range
+    extreme_mask = np.where(np.logical_or(norm_vals < extreme_vals[0], norm_vals > extreme_vals[1]))
```
Comment (ngreenwald) [inline src/toffy/normalize.py:741]: What are the `extreme_vals` currently set to?

**Review by ngreenwald — CHANGES_REQUESTED** [review]

File: src/toffy/normalize.py (hunk 1 above)
Comment (alex-l-kong) [reply src/toffy/normalize.py:741]: `(0.4, 1.1)`. Had been meaning to ask if we should adjust these.

File: src/toffy/normalize.py [hunk 2]
```diff
@@ -718,12 +718,35 @@ def normalize_fov(img_data, norm_vals, norm_dir, fov, channels, extreme_vals):
...
+            "The following channel(s) will suffer a >10X increase after normalization "
+            "for fov {}. Normalization capped at 10X for: {}.".format(fov, abnormal_increase_chans)
+        )
+        norm_vals[abnormal_increase_mask] = 0.1
+
+    decrease_mask = np.where(norm_vals > 1)
+    if len(decrease_mask) > 0:
+        decrease_chans = np.array(channels)[decrease_mask]
+        warnings.warn(
+            "The following channel(s) will suffer a decrease after normalization "
```
[the remark this answers is not in the document]
Comment (alex-l-kong) [reply src/toffy/normalize.py:735]: Per today's stand up we'll leave the normalization capped at 1.

File: src/toffy/normalize.py (hunk 1 above)
Comment (alex-l-kong) [reply src/toffy/normalize.py:741]: @ngreenwald do we want to change the `extreme_vals` or is `(0.4, 1.1)` good?

File: src/toffy/normalize.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ngreenwald) [reply src/toffy/normalize.py:735]: Let's cap the correction at 1.1. So anything with a correction value larger than 1.1 gets a correction of 1.1

[the remark this answers is not in the document]
Comment (ngreenwald) [reply src/toffy/normalize.py:735]: And anything with a correction value less than 0.1 still gets 0.1.

File: src/toffy/normalize.py (hunk 1 above)
Comment (ngreenwald) [reply src/toffy/normalize.py:741]: Yes, let's keep it here. Then add an additional warning for anything less than 0.1 saying it was capped at 0.1, and that the counts may be outside the range where normalization works

Comment (alex-l-kong) [reply src/toffy/normalize.py:741]: Ok sounds good. We can probably include this inside the `warning` statement which indicates the values got capped at 10X normalization (aka norm_val capped at 0.1).

File: src/toffy/normalize.py [hunk 3]
```diff
@@ -718,12 +718,37 @@ def normalize_fov(img_data, norm_vals, norm_dir, fov, channels, extreme_vals):
...
+            "Note that the counts for these channels may be out of range, "
+            "please check your normalization settings.".format(fov, abnormal_increase_chans)
+        )
+        norm_vals[abnormal_increase_mask] = 0.1
+
+    decrease_mask = np.where(norm_vals > 1.1)[0]
+    if len(decrease_mask) > 0:
+        decrease_chans = np.array(channels)[decrease_mask]
+        warnings.warn(
+            "The following channel(s) will suffer a significant decrease after normalization "
```
Comment (ngreenwald) [inline src/toffy/normalize.py:737]:
```suggestion
            "The following channel(s) are at greater than 110% sensitivity, which may indicate that your detector was overgained for this run, "
```

**Review by ngreenwald — APPROVED** [review]
Looks good, see wording change below

**Pull request merged**